Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1497,7 +1488,368 @@ The hooked function outputted the decrypted string. You extracted the secret str
 
 You've now covered the basics of static/dynamic analysis on Android. Of course, the only way to *really* learn it is hands-on experience: build your own projects in Android Studio, observe how your code gets translated into bytecode and native code, and try to crack our challenges.
 
-In the remaining sections, we'll introduce a few advanced subjects, including kernel modules and dynamic execution.
+In the remaining sections, we'll introduce a few advanced subjects, including process exploration, kernel modules and dynamic execution.
+
+##### Process Exploration
+
+When testing an app, process exploration can provide the tester with deep insights into the app process memory. It can be achieved via runtime instrumentation and allows to perform tasks such as:
+
+- Retrieving the memory map and loaded libraries.
+- Searching for occurrences of certain data.
+- After doing a search, obtaining the location of a certain offset in the memory map.
+- Performing a memory dump and inspect or reverse engineer the binary data _offline_.
+- Reverse engineering a native library while it's running.
+
+As you can see, these tasks are rather supportive and/or passive, they'll help us collect data and information that will support other techniques. Therefore, they're normally used in combination with other techniques such as method hooking.
+
+In the following sections you will be using r2frida to retrieve information straight from the app runtime. First start by opening an r2frida session to the target app (e.g. [HelloWorld JNI](https://github.com/OWASP/owasp-mstg/raw/master/Samples/Android/01_HelloWorld-JNI/HelloWord-JNI.apk "HelloWorld JNI") APK) that should be running on your Android phone (connected per USB). Use the following command:
+
+```bash
+$ r2 frida://usb//sg.vantagepoint.helloworldjni
```

Comment:
where do we explain on how to get r2 frida working? (e.g. a link to the readme for installation instructions)

Reply:
done